src/installer/corehost/cli/hostmisc/pal.unix.cpp

```diff
@@ -583,20 +585,69 @@ pal::string_t pal::get_current_os_rid_platform()
 pal::string_t pal::get_current_os_rid_platform()
 {
     pal::string_t ridOS;
-
     char str[256];
-
     size_t size = sizeof(str);
     int ret = sysctlbyname("kern.osrelease", str, &size, NULL, 0);
+
     if (ret == 0)
     {
-        char *pos = strchr(str,'.');
+        char *pos = strchr(str, '.');
+        if (pos)
+        {
+            ridOS.append(_X("freebsd."))
+                 .append(str, pos - str);
+        }
+    }
+
+    return ridOS;
+}
+#elif defined(__sun)
+pal::string_t pal::get_current_os_rid_platform()
```

Comment:
If I remember correctly, there was a time where we were trying very hard to avoid having special case mapping in the RID calculation.  It means folks will need a new copy of the RID algorithm in order to get the right result on these platforms.  It will also mean the old algorithm will stop returning the truth on these platforms.  @eerhardt

Reply:
The way the new 5.0 API [`RuntimeInformation.RuntimeIdentifier`](https://github.com/dotnet/runtime/blob/0c043d58ccce29e1d22acd4f77b0e0b352579ac5/src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/RuntimeInformation.cs#L50-L51) is implemented is it just returns the RID from the host. So as long as any users are using this new API, they will get the correct RID.

If users are using the deprecated [Microsoft.DotNet.PlatformAbstractions](https://www.nuget.org/packages/Microsoft.DotNet.PlatformAbstractions/) library, they won't get the correct RID on these systems, and will need to update to the new API.

Reply:
cc @vitek-karas @swaroop-sridhar - for the host changes.

Reply:
@eerhardt, does the deprecated PlatformAbstractions library provides certain feature which RuntimeIndentifier does not? I found that dotnet/sdk still using it in the following three files:

```sh
sdk$ git grep -li PlatformAbstractions
src/Cli/Microsoft.DotNet.Cli.Utils/WindowsRegistryEnvironmentPathEditor.cs
src/Layout/redist/targets/GenerateLayout.targets
src/Tasks/Microsoft.NET.Build.Tasks/ProjectContext.cs
```

Generally, I think it would be a better model, if we don't have to update platforms in SDK repo separately each time a new platform is introduced in runtime repo. For example, roslyn, aspnet, even msbuild and many other repos don't know/care how many OS platforms are supported by runtime, they just work everywhere (MSIL ftw). At least SDK should not block the new platforms by default (especially if that hardcoded platform list is just used for telemetry). :)

Reply:
> I found that dotnet/sdk still using it in the following three files:

In the two .cs files, it is only being used by a `using` statement. This was a left-over of old code, and can be removed. The reason the namespace still exists is because another library (Microsoft.Extensions.DependencyModel) has an Obsolete type in that namespace.

For `src/Layout/redist/targets/GenerateLayout.targets`, this was actually telling the build system to not include the PlatformAbstractions assembly coming from vstest. vstest doesn't reference that library anymore, so it can be removed as well.

I've removed these with https://github.com/dotnet/sdk/pull/11857.

> Generally, I think it would be a better model, if we don't have to update platforms in SDK repo separately each time a new platform is introduced in runtime repo. 

> At least SDK should not block the new platforms by default (especially if that hardcoded platform list is just used for telemetry). :)

Can you open an issue in dotnet/sdk for this? Note that it isn't just telemetry - but also the SDK prints OS platform information in `dotnet --info`.